build_tools/build_wheels_env.sh

```diff
@@ -0,0 +1,29 @@
+# The wheels are built on a centos docker image
```

Comment:
Could you please explain when and why this is necessary? Don't you in `build_package.sh` install swig3 via the package manager?

Reply:
THanks for the more detailed comment, but I don't understand why there are now two SWIG installations:

1. There's one in `build_wheels_env.sh`
2. and a 2nd in `build_package.sh`.
 
Isn't `build_wheels_env.sh` executed first so that we have a SWIG installation and don't need the extra installation in `build_package.sh`?

Reply:
Ok, the first installation of swig is, as you said, one needed for the package to be installed. We install the package to create a distribution for example. The question is why we have the second SWIG installation. 

If you see the second swig, which is in `build_wheels_env.sh`, it is used for another distro, which is centos. I would have gladly used the same file, but we have to deal with yum support.

Why centos? To build wheels for many linux packages  as described [here](https://cibuildwheel.readthedocs.io/en/stable/options/#manylinux-image) -- particularly we use this cibuildwheel package that uses images from quay.io.  In this docker image, we have to make sure swig is there too.

Reply:
That all makes sense. So the reason we call `env.sh` inside `build_package.sh` is because `build_package.sh` is by both `build_wheels.sh` and `build_source.sh` and it is needed for `build_source.sh` (as far as I see it's not needed for `build_wheels.sh` as the environment is already installed by `build_wheels_env.sh`)? Would it then be possible to move the call to `env.sh` to `build_source.sh` to make this clearer?

Reply:
Sorry for the trouble, let me elaborate on this:
> Ok, the first installation of swig is, as you said, one needed for the package to be installed. 

For both, building wheels and building a distribution we need the compiled pyrfr package. That is, the following steps are required:

1. Create a python directory that mimics a traditional python package directory:
```
ls python_package:
build     how_to_release.txt  LICENSE_cereal-1.2.2  pyrfr     tests
examples  include             MANIFEST.in           setup.py
```
2. From this directory, we can create a distribution via `python setup.py sdist`  and wheels by using  `python -m cibuildwheel`.

For point 2, the 2 jobs in github actions, that create a wheel and the other that create dist,  are independent, so they have their own directory structure. Each one of then have to create this python package directory structure to work. Build package creates this structure.

We can rename the files to make it more intuitive with steps? Like step1_build_python_package_directory.sh, and so on?

Reply:
Okay, we resolved this offline. This was necessary because creating a binary distribution was not done inside the wheel building container, but outside. Now that the compilation was moved into the wheel building container via 7c50488 there are no longer two install of swig.